pkg/container-utils/docker/docker.go

```diff
@@ -113,7 +108,25 @@ func (c *DockerClient) GetContainer(containerID string) (*runtimeclient.Containe
 
 	containerJSON, err := c.client.ContainerInspect(context.Background(), containerID)
 	if err != nil {
-		return nil, err
```

Comment:
The same situation can happen using the CRI, I reproduced it with containerd.

I used the container-collection example (modified to print verbose messages) to try to capture the creation of the container:

```bash
$ cd examples/container-collection/; go build .; sudo ./container-collection
Container added: "kube-apiserver" pid 1847
Container added: "calico-kube-controllers" pid 4940
Container added: "gadget" pid 283168
...
# I created the container here
...
DEBU[0007] Runtime enricher (containerd): failed to get container: rpc error: code = NotFound desc = an error occurred when try to find container "cont1": not found
Container added: "" pid 308477
```

I run the container like this:
```bash
$ sudo ctr run --rm -t docker.io/library/nginx:1.21 cont1
```